operator/pkg/util/k8s.go

```diff
@@ -68,3 +72,32 @@ func DetectSupportedJWTPolicy(config *rest.Config) (JWTPolicy, error) {
 func GKString(gvk schema.GroupKind) string {
 	return fmt.Sprintf("%s/%s", gvk.Group, gvk.Kind)
 }
+
+// ValidateIOPCAConfig validates if the IstioOperator CA configs are applicable to the K8s cluster
+func ValidateIOPCAConfig(client kubernetes.Interface, iop *iopv1alpha1.IstioOperator) error {
```

Comment:
Nit: suggest to update `ValidateIOPCAConfig()` to accept `rest.Config` as parameter, similar to what we do in the `DetectSupportedJWTPolicy()`, see: 
https://github.com/istio/istio/blob/2bef9460f448242bc49ade969037cd5b7205ac68/operator/pkg/util/k8s.go#L33

Then we can pass the `config` of `ReconcileIstioOperator` object to `ValidateIOPCAConfig()`, we don't need to regenerate another client in the operator controller.

Reply:
Unsure about this - This function takes `kubernetes.Interface` as an argument because I can't write unit tests for this without it. In order to "fake" a server version (used inside this function), I need to pass the faked `kubernetes.Client` into the function with a fake server version. It would be impossible to achieve the same functionality using a `rest.Config` since it doesn't appear to any field to indicate the server version.